src/main/java/com/wavefront/opentracing/WavefrontSpan.java

```diff
@@ -48,9 +54,14 @@
   // Store it as a member variable so that we can efficiently retrieve the component tag.
   private String componentTagValue = NULL_TAG_VAL;
 
+  private static Set<String> SINGLE_VALUED_TAG_KEYS = new HashSet<>(Arrays.asList(
+      Constants.APPLICATION_TAG_KEY, Constants.SERVICE_TAG_KEY, Constants.CLUSTER_TAG_KEY,
+      Constants.SHARD_TAG_KEY));
+
   WavefrontSpan(WavefrontTracer tracer, String operationName, WavefrontSpanContext spanContext,
                 long startTimeMicros, long startTimeNanos, List<Reference> parents,
-                List<Reference> follows, List<Pair<String, String>> tags) {
+                List<Reference> follows, List<Pair<String, String>> tags,
+                List<Pair<String, String>> globalTags) {
```

Comment:
We already have access to the `tracer` instance. Can just call `tracer.getTags()` within the constructor rather than pass this in.